### PR TITLE
Remove support for AppExeCLinks to retrieve target

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

Remove use of undocumented APIs for AppExeCLinks.  Per AppX team, they have no intent of publishing structure and says users should not directly work with the target.

New behavior which seems acceptable:

image

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by iSazonov** [comment]
Is IO_REPARSE_TAG_APPEXECLINK const itself public?

**Comment by vexx32** [comment]
Rather than a blank target, maybe we could insert a token such as `<AppExeLink>` to indicate what it is, so that at least users have that information and don't try to work with it like a normal symlink?

**Comment by iSazonov** [comment]
Really there can be _many_ reparse point tags for which .Net API return null target. So I think it is better to follow .Net API behavior.

See also https://github.com/dotnet/runtime/pull/58233#issuecomment-911971904

**Comment by SteveL-MSFT** [comment]
@iSazonov that const is in winnt.h:

image

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs [hunk 1]
```diff
@@ -8506,16 +8490,6 @@ private static string WinInternalGetTarget(SafeFileHandle handle)
                         targetDir = Encoding.Unicode.GetString(reparseMountPointDataBuffer.PathBuffer, reparseMountPointDataBuffer.SubstituteNameOffset, reparseMountPointDataBuffer.SubstituteNameLength);
                         break;
 
-                    case IO_REPARSE_TAG_APPEXECLINK:
```
Comment (daxian-dbw) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs:8509]: The const string `IO_REPARSE_TAG_APPEXECLINK` is not used anywhere else. Better to remove it.

Comment (SteveL-MSFT) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs:8509]: It's still used on line 8289 as a check for OneDrive reparse points

Comment (daxian-dbw) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs:8509]: Right, I missed that :)

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Review by iSazonov — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**